### PR TITLE
Add Hangman challenge

### PR DESCRIPTION
**Pull request by LeoCumpli21**
## Summary

This PR closes LeoCumpli21/PyZombis#13.

I implemented the Hangman game from the original course. I did it as a progressive exercise, where most activecodes need the previous ones to be completed before them to work appropriatly. 

### About the exercise
* It is thought as being a challenge more than a simple exercise.

### About the `Playwright` test:
* It is a big test due to the fact that the activecodes are linked to each other with the `:include: ` parameter.
* In the video below, sometimes the screen seems to stall. This happens because the *dialog prompts* are being handled, but Playwright doesn't show this.

## Checklist

- [x] Variables, functions and comments are translated to Spanish
- [x] Functions follow underscore notation
- [x] Spell check done & typos fixed
- [x] All python code is PEP8 compliant
- [x] Test coverage with Playwright implemented; locators are Pyhton code
- [x] Reviewers assigned (all peers & at least 1 mentor)

## Screenshots

https://user-images.githubusercontent.com/80920858/126044041-6bc53309-b740-42dc-96ca-faaf01aa29e0.mp4

### CODE REVIEW

**Comment by LeoCumpli21** [comment]
I use webkit as the browser for the playwright test. This is why locally this final version didn't gave me any problems. However, I didn't realize that with chrome Playwright has trouble accepting the dialog prompt.

**Comment by LeoCumpli21** [comment]
> I use webkit as the browser for the playwright test. This is why locally this final version didn't gave me any problems. However, I didn't realize that with chrome Playwright has trouble accepting the dialog prompt, hence making the test fail.

Done! Now it works on Chrome as well.

File: _sources/index.rst [hunk 1]
```diff
@@ -46,6 +46,7 @@ Contenidos:
    quiz/Quiz14.rst
    quiz/QuizExtras.rst
    quiz/QuizExtras2.rst
+   lectures/Reto01.rst
```
Comment (angelasofiaremolinagutierrez) [inline _sources/index.rst:49]: I suggest to create a different folder for these challenges you are creating. Something like: `retos/Reto01.rst`

File: _sources/lectures/Reto01.rst [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        
+        Ya ha definido todas las funciones y variables necesarias para el programa. Si ha llegado hasta aquí, 
+        debió haber cumplido con todo y haber pasado todas las pruebas unitarias. De ser así, puede correr el
+        siguiente programa para probar su juego.
+
+        .. activecode:: ac_r01_8
+            :nocodelens:
+            :include: ac_r01_1, ac_r01_2, ac_r01_3, ac_r01_4, ac_r01_5, ac_r01_6, ac_r01_7
+
+            # Esto es para aumentar el tiempo de ejecución de Runestone
```
Comment (angelasofiaremolinagutierrez) [inline _sources/lectures/Reto01.rst:285]: Is it possible to hide these lines from the activecode block? Maybe in the way we hide the tests. I don't think it's necessary for the student to see them.

**Review by angelasofiaremolinagutierrez — APPROVED** [review]
LGTM! just a couple of notes ;)

File: _sources/lectures/Reto01.rst [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            :nocodelens:
+
+            Para poder jugar ahorcado necesitamos palabras para adivinar. En este caso vamos a 
+            recuperar esas palabras de la API de University Domains. Estas palabras serán ciudades
+            del mundo. 
+            Con la librería ``requests`` acceda a la API de University Domains. La url es la siguiente 
+            http://universities.hipolabs.com/search 
+            Desarrolle la función ``escoger`` que escoja de manera **aleatoria** una ciudad dentro de las ciudades de la 
+            lista ``ciudades``. Esta ciudad será devuelta por la función.
+
```
Comment (NicolasSandoval) [inline _sources/lectures/Reto01.rst:106]: Perhaps here we could clarify that the names of the cities are in English

File: _sources/lectures/Reto01.rst [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+            ~~~~
+            import requests
+            import json
+            # Debe importar otra librería
```
Comment (NicolasSandoval) [inline _sources/lectures/Reto01.rst:110]: We could add another valid option, like: 
```python3
import random

random.choice(ciudades)
```

File: _sources/lectures/Reto01.rst [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+            :nocodelens:
+        
+            Desarrolle la función ``adivinar``. Recibe como parámetro una cadena (``letras``) con todas las letras ya 
+            probadas (correctas + incorrectas). Devuelve una letra minúscula que no se ha probado antes. 
+            Si el usuario ingresa más de una letra u otro carácter especial, se debe imprimir un mensaje 
+            alertando al usuario de su error. Por ejemplo: ``"Caracter inválido"``
+            Esta función tendrá un bucle que hará que todos los datos sean coherentes y solo pasará 
+            a tu programa principal la letra minúscula que no se probó antes. Guarde lo ingresado por el usuario en la
+            variable ``x``. **Nota**: La función se va a ejecutar una vez para verificar si pasa las pruebas unitarias.
+
```
Comment (NicolasSandoval) [inline _sources/lectures/Reto01.rst:178]: Here we must add that the code needs to request the user to input a letter.

File: _sources/lectures/Reto01.rst [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+        Hagamos posible que el usuario decida si quiere volver a jugar o no.
+
+        .. activecode:: ac_r01_6
+            :nocodelens:
+
+            Desarrolle la función ``jugar_nuevamente`` que pregunte al usuario si quiere volver a 
+            jugar. La respuesta debe ser una *s* o *n* (S/N). Sin importar si el usuario ingresa la respuesta
+            en mayúscula o minúscula, el programa debe transformarla a minúscula. La función devuelve ``True`` o ``False``
+            dependiendo de la respuesta. **Nota**: La función se va a ejecutar dos vez para verificar si pasa las pruebas unitarias.
```
Comment (NicolasSandoval) [inline _sources/lectures/Reto01.rst:218]:
```suggestion
            dependiendo de la respuesta. **Nota**: La función se va a ejecutar dos veces para verificar si pasa las pruebas unitarias.
```

**Review by NicolasSandoval — CHANGES_REQUESTED** [review]
LGTM! I left a few comments and suggestions.

File: _sources/index.rst (hunk 1 above)
Comment (LeoCumpli21) [reply _sources/index.rst:49]: Sure! In another PR I will tackle this 😄 Thanks for the suggestion.

File: _sources/lectures/Reto01.rst (hunk 2 above)
Comment (LeoCumpli21) [reply _sources/lectures/Reto01.rst]: I tried to hide those lines, but didn't work.

**Review by NicolasSandoval — APPROVED** [review]
Great work! Thanks.

**Review by y-martinez — APPROVED** [review]
LGTM 🚀

**Pull request merged**